assets/js/modules/analytics-4/components/widgets/TopDeviceDrivingPurchasesWidget.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * TopDeviceDrivingPurchases component.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useSelect, useInViewSelect } from 'googlesitekit-data';
+import MetricTileText from '../../../../components/KeyMetrics/MetricTileText';
+import {
+	CORE_USER,
+	KM_ANALYTICS_TOP_DEVICE_DRIVING_PURCHASES,
+} from '../../../../googlesitekit/datastore/user/constants';
+import {
+	DATE_RANGE_OFFSET,
+	MODULES_ANALYTICS_4,
+} from '../../datastore/constants';
+import { numFmt } from '../../../../util';
+import { get } from 'lodash';
+import whenActive from '../../../../util/when-active';
+import ConnectGA4CTATileWidget from './ConnectGA4CTATileWidget';
+
+function TopDeviceDrivingPurchases( { Widget } ) {
+	const dates = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+			compare: true,
+		} )
+	);
+
+	const totalPurchasesReportOptions = {
+		...dates,
+		metrics: [
+			{
+				name: 'ecommercePurchases',
+			},
+		],
+	};
+
+	const deviceReportOptions = {
+		...dates,
+		dimensions: [ 'deviceCategory' ],
+		metrics: [
+			{
+				name: 'ecommercePurchases',
+			},
+		],
+		limit: 1,
+		orderBy: 'ecommercePurchases',
+	};
+
+	const totalPurchasesReport = useInViewSelect(
+		( select ) =>
+			select( MODULES_ANALYTICS_4 ).getReport(
+				totalPurchasesReportOptions
+			),
+		[ totalPurchasesReportOptions ]
+	);
+
+	const deviceReport = useInViewSelect(
+		( select ) =>
+			select( MODULES_ANALYTICS_4 ).getReport( deviceReportOptions ),
```

Comment:
Also, we can omit this report request if there is no data for `totalPurchasesReport`. So we can include additional condition alongside the one above if event is not detected any more.

Besides, the zero data currently looks like this:

<img width="308" alt="image" src="https://github.com/user-attachments/assets/9e06487e-470e-482c-a238-6df92098bf3f">

The device should be showing `-` instead of device name, skipping the `deviceReport` and returning `undefined` when we know there is no data, will fix this. 

Have in mind that for this metric `totalPurchasesReport` will always return `rows`, unlike for other reports, where rows would be empty, so we can't use that to check if report has any data. One solution is to check if the value of the result is greater then `0`, like here https://github.com/google/site-kit-wp/blob/9a1a4f1d66be97d1839e663b4c3b4eceb839d081/assets/js/modules/analytics-4/components/widgets/TopTrafficSourceDrivingPurchasesWidget.js#L96-L99